h5py/h5pl.pyx

```diff
@@ -0,0 +1,57 @@
+# This file is part of h5py, a Python interface to the HDF5 library.
+#
+# http://www.h5py.org
+#
+# Copyright 2008-2019 Andrew Collette and contributors
+#
+# License:  Standard 3-clause BSD; see "license.txt" for full license terms
+#           and contributor agreement.
+
+"""
+    HDF5 plugin interface.
+"""
+
+include "config.pxi"
+from utils cimport emalloc, efree
+
+# === C API ===================================================================
+
+IF HDF5_VERSION >= (1, 10, 1):
+    cpdef append(const char* search_path):
```

Comment:
I'm not very familiar with the low-level API, but should these have the `with_phil` decorator for locking? Check with the other maintainers before adding this;  it might not be necessary.

Reply:
the `phil` lock is mostly about making sure that we don't create / reuse / destroy c++ objects in an unsafe way (this is to support the correct ref-counting on holding c++ File objects open even if the user drops all (direct) python references to the File and only keeps Datasets).  See the code in `_objects.pyx` which is rather racey (including an issue where we discovered we have to turn off gc temporarily due to `list(of_a_dict)` having "mutation during iteration" issues).

That is a very long way of saying I don't think we need to lock with the same lock as everything else (as if the user has race conditions between adding plugins to the search path and opening files I think that is on them).

On the other hand, these should be used rarely enough that locking could not hurt.  We may also want to lock them between eachother (as I can imagine an `append` and a `remove` colliding and ending up with a "gap" in the hdf5 datastrucutres....).